ddtrace/internal/telemetry/writer.py

```diff
@@ -188,10 +187,8 @@ def add_integration(self, integration_name, auto_enabled):
         :param str integration_name: name of patched module
         :param bool auto_enabled: True if module is enabled in _monkey.PATCH_MODULES
         """
-        with self._lock:
-            if self._enabled is not None and not self._enabled:
-                return
-
+        if self._enabled is None or self._enabled:
```

Comment:
Out of curiosity, can you explain why removing the `if self._enabled is None` resulted in application issues (failed gunicorn tests)? Thanks!